### DISCUSSION TITLE
Downloading without zipping

### DISCUSSION THREAD

**Discussion by Vulkandr** [rejetto/hfs#126] [Q&A] [answered]
Is it possible to add a download button next to zip button or the file itself? is that the point of the download manager plugin? as i cant seem to get it to work. downloading through external download managers with the link works. because if the browser supports the filetype itl just open it and then you have to save it from there, which is less obvious for less tech savvy users and ive got a few like that lol. maybe its possible through editing the template? i only know just enough html to make my own basic skin with a background, but i wouldnt know how to add that functionality in. if this is planned to be added, my bad, i understand its still early days.

Thanks

**Comment by rejetto** [comment, marked as answer]
you are supposed to download files just by clicking on their names.
As you say, some files are just opened, and that's a consequence of the "mime" configuration.
You are right, not everybody knows that you can right click on a file and save.

I'm not sure what a good solution is.
There are 2 problems here:

A) we have 2 possible actions (open and save), so the user should be able to choose, while right now there is only 1 link to click.
Should I add a button next to each file?
If yes, should the button be for open or save?

B) Not every file can be open, and I don't really know what files can be opened by YOUR browser.
This means that if you show you an "open" button, and then it is not open, you are probably pissed off.

Maybe best solution is to offer an explicit download button, very small, in form of an icon.
Clicking the name would be a "try to open", but since it's just the name of the file, no explicit promise is done.

**Reply by Vulkandr** [reply]
What you said last makes the most sense to me. Im assuming the file not being openable would still result in a 'save' scenario like it does now? If so, that would keep current functionality as is for those that want it, while adding the option to only save if one so chooses, basically nothing would be lost.
